airbyte-integrations/connectors/source-kapiche-export-api/source_kapiche_export_api/source.py

```diff
@@ -0,0 +1,356 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple, Union
+
+import requests
+import pandas as pd
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth.core import HttpAuthenticator
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+
+# Basic full refresh stream
+class KapicheExportApiStream(HttpStream, ABC):
+    """
+    Abstract Stream class for kapiche-export-api connector.
+    All streams from airbyte must inherit from this as a base class.
+    """
+    
+    def __init__(self, authenticator: HttpAuthenticator = None):
+        super().__init__(authenticator)
+        self._url_base = ''
+
+    @property
+    def url_base(self) -> str:
+        """
+        :return: URL base for the  API endpoint e.g: if you wanted to hit https://myapi.com/v1/some_entity then this should return "https://myapi.com/v1/"
+        """
+        return self._url_base
+
+    @url_base.setter
+    def url_base(self, value:str):
+        self._url_base =  value
+
+    
+
+
+class ExportDataGet(KapicheExportApiStream, ABC):
+    """
+    TODO: Change class name to match the table/data source this stream corresponds to.
+    """
+
+    primary_key = None
+    cursor_field = 'document_id__'
+    
+    # needed to instantiate the class
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
```

Comment:
Would this be

```suggestion
    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, str]]:
```

?